lib/vrl/stdlib/src/keys.rs

```diff
@@ -0,0 +1,75 @@
+use ::value::Value;
+use vrl::function::ArgumentList;
+use vrl::function::Compiled;
+use vrl::function::Example;
+use vrl::function::FunctionCompileContext;
+use vrl::prelude::*;
+use vrl::state::TypeState;
+use vrl::Expression;
+use vrl::Function;
+
+fn keys(value: Value) -> Resolved {
+    let mut vec: Vec<Value> = Vec::new();
+    let value_btree = value.try_object()?;
+
+    for (k, _v) in value_btree {
+        vec.push(Value::Bytes(Bytes::from(k.clone())))
+    }
+    Ok(Value::Array(vec))
```

Comment:
Your solution works works well and is very straight-forward!

However, we could consider using the [`BTreeMap::into_keys`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.into_keys), [`From::<String>::from` for `Value`](https://github.com/vectordotdev/vector/blob/9db87b27af5b1f057ce7759042d26046cc3cebc1/lib/value/src/value/convert.rs#L281-L285) and [`Iterator::collect`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect) methods from the standard library here and let the compiler infer the types for us.

As an additional bonus, we don't need to `clone` the key as we consume the iterator:

```suggestion
    let object = value.try_object()?;
    let keys = object.into_keys().map(Value::from);
    Ok(Value::Array(keys.collect()))
```